### PR TITLE
ah - use new GT encoding

### PR DESCRIPTION
**Pull request by ahaessly**
change GTs to single character, drop hom ref, add sample metrics to sample metadata tsv

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/variantdb/arrays/ArraySampleTsvCreator.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            String columns = null;
+            String values = null;
+            reader = new BufferedReader(new FileReader(metricsFilepath));
+            String line = reader.readLine();
+            while (line != null) {
+                if (!line.startsWith("#") && !line.trim().isEmpty()) {
+                    if (columns == null) {
+                        columns = line;
+                    } else {
+                        values = line;
```
Comment (meganshand) [inline src/main/java/org/broadinstitute/hellbender/tools/variantdb/arrays/ArraySampleTsvCreator.java:33]: Is there only one line in the metrics file? If so, can you put in a check for that?

File: src/main/java/org/broadinstitute/hellbender/tools/variantdb/arrays/RawArrayFieldEnum.java [hunk 2]
```diff
@@ -30,67 +36,39 @@ public String getColumnValue(VariantContext variant, ProbeInfo probeInfo, String
 
     // This where the validation step (required vs not) lives  -- fail if there is missing data for a required field
```
Comment (meganshand) [inline src/main/java/org/broadinstitute/hellbender/tools/variantdb/arrays/RawArrayFieldEnum.java:37]: Should this comment be deleted too?

**Review by meganshand — COMMENTED** [review]
Looks good to me, just a few questions.

**Review by meganshand — APPROVED** [review]
Looks good, merge when tests pass.

File: src/main/java/org/broadinstitute/hellbender/tools/variantdb/CommonCode.java [hunk 3]
```diff
@@ -1,29 +1,51 @@
...
 
 //TODO rename this or get rid of it. a place holder for now
 public class CommonCode {
-    public static final String NORMX = "NORMX";
-    public static final String NORMY = "NORMY";
-    public static final String BAF = "BAF";
-    public static final String LRR = "LRR";
 
 
+    public static String getGTString(final VariantContext variant) {
```
Comment (kcibul) [inline src/main/java/org/broadinstitute/hellbender/tools/variantdb/CommonCode.java:22]: I'm so surprised this code doesn't exist anywhere?  the VCFWriter must be doing (largely) the same thing?

File: src/main/java/org/broadinstitute/hellbender/tools/variantdb/arrays/RawArrayFieldEnum.java [hunk 4]
```diff
@@ -3,126 +3,96 @@
...
-        public String getColumnValue(final VariantContext variant, ProbeInfo probeInfo, String sampleId) {
-            Genotype g = variant.getGenotype(0);
             RawArrayTsvCreator.GT_encoding gt = RawArrayTsvCreator.GT_encoding.MISSING;
-            if (g.isHomRef() || g.isHomVar()) {
-                Allele allele = g.getAllele(0);
-                if (allele.basesMatch(probeInfo.alleleA)) {
-                    gt = RawArrayTsvCreator.GT_encoding.AA;
-                } else if (allele.basesMatch(probeInfo.alleleB)) {
-                    gt = RawArrayTsvCreator.GT_encoding.BB;
+            if (alleleIndexes.size() == 2) {
```
Comment (kcibul) [inline src/main/java/org/broadinstitute/hellbender/tools/variantdb/arrays/RawArrayFieldEnum.java:38]: what if it's not two?  should we throw an exception?

File: src/main/java/org/broadinstitute/hellbender/tools/variantdb/arrays/RawArrayFieldEnum.java [hunk 5]
```diff
@@ -3,126 +3,96 @@
...
+                    if (uniqueAlleleIndexes.containsAll(new HashSet<>(Arrays.asList(0, 1)))) {
+                        gt = RawArrayTsvCreator.GT_encoding.HET0_1;
+                    } else if (uniqueAlleleIndexes.containsAll(new HashSet<>(Arrays.asList(1, 2))))
+                        gt = RawArrayTsvCreator.GT_encoding.HET1_2;
                 }
-            } else if (g.isHet()) {
-                gt = RawArrayTsvCreator.GT_encoding.AB;
             }
-            return gt.getValue();
+            return gt == RawArrayTsvCreator.value_to_drop ? "null" : gt.getValue();
```
Comment (kcibul) [inline src/main/java/org/broadinstitute/hellbender/tools/variantdb/arrays/RawArrayFieldEnum.java:58]: do you mean the string `null` or an actual null?

**Review by kcibul — APPROVED** [review]
Left a few comments

File: src/main/java/org/broadinstitute/hellbender/tools/variantdb/CommonCode.java (hunk 3 above)
Comment (ahaessly) [reply src/main/java/org/broadinstitute/hellbender/tools/variantdb/CommonCode.java:22]: yes, but it is very embedded in the actual writing of the output and in htsjdk which for some reason i think it's more difficult for us to modify to make it modular.

File: src/main/java/org/broadinstitute/hellbender/tools/variantdb/arrays/RawArrayFieldEnum.java (hunk 4 above)
Comment (ahaessly) [reply src/main/java/org/broadinstitute/hellbender/tools/variantdb/arrays/RawArrayFieldEnum.java:41]: added a warning

File: src/main/java/org/broadinstitute/hellbender/tools/variantdb/arrays/RawArrayFieldEnum.java (hunk 5 above)
Comment (ahaessly) [reply src/main/java/org/broadinstitute/hellbender/tools/variantdb/arrays/RawArrayFieldEnum.java:63]: yes, the actual string "null". the tsv needs some value and the bq import will convert it to an actual null in the db. added to the comment to explain

**Pull request merged**